backend/utils/authutils/encryption.go

```diff
@@ -0,0 +1,47 @@
+package authutils
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"errors"
+)
+
+func EncryptData(data, key []byte) ([]byte, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCMWithRandomNonce(block)
+	if err != nil {
+		return nil, err
+	}
+
+	ciphertext := gcm.Seal(nil, nil, data, nil)
+	return ciphertext, nil
+
+}
+
+func DecryptData(ciphertext, key []byte) ([]byte, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCMWithRandomNonce(block)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ciphertext) < gcm.NonceSize() {
```

Comment:
instead of checking only against `gcm.NonceSize()` it might be better to perform the following check:

```golang
// if (ciphertext too large) || (ciphertext too small)
if (len(ciphertext) > (constants.KeyLength + gcm.Overhead())) || len(ciphertext) - (constants.KeyLength + gcm.NonceSize()) < 0 {
}
```

this is under the assumption that we only call `DecryptData` to decrypt the database key, whose plaintext length in bytes is given by `constants.KeyLength`. probably a very good idea to verify this change using tests tho, easy to be off by a few bytes ^^